delete_student.py

```diff
@@ -0,0 +1,25 @@
+def random_st_delete(list_of_student):
+    for i in range(len(list_of_student)):
+        print(i+1, list_of_student[i])
+    print('are all students here? Answer y/n')
+    x = input()
+    if x == 'n':
+        print('who is absent today? print their №')
+        absent_student = [int(i) for i in input().split()]
+        for j in range(len(absent_student)):          
+            list_of_student.remove(list_of_student[j])
+    while len(list_of_student) > 0:
+        print('Choose a random student? y/n')
+        if input() == 'y':
+            q = random.choice(list_of_student)
+            print(q)
+            list_of_student.remove(q)
+
+if __name__ == '__main__':
+    list_of_student = [line.strip().split("\t",1)[1] for line in open("../data_input/list_of_students.txt").readlines()]
```

Comment:
There is no guarantee that the file will be closed in a list comprehension (see [this](https://stackoverflow.com/questions/18840880/will-using-list-comprehension-to-read-a-file-automagically-call-close) for details). Please, use a context manager.

Reply:
In Windows I also have the following exception:
```
Traceback (most recent call last):
  File "C:/Users/Anton.Bragin/PycharmProjects/StudentsProject/delete_student.py", line 21, in <module>
    list_of_student = [line.strip().split()[1] for line in open("data_input/list_of_students.txt").readlines()]
  File "C:\Users\Anton.Bragin\Envs\py36\lib\encodings\cp1251.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 18: character maps to <undefined>
```